plugins/amoeba/platforms/common/src/kernels/amoebaGk.cc

```diff
@@ -46,51 +76,184 @@ KERNEL void computeSurfaceAreaForce(GLOBAL mm_long* RESTRICT bornForce, GLOBAL m
 typedef struct {
     real3 pos;
     real bornSum;
-    float radius, scaledRadius, padding;
+    float radius, scaleFactor, descreenRadius, neckFactor, padding;
 } AtomData1;
 
-DEVICE real computeBornSumOneInteraction(AtomData1 atom1, AtomData1 atom2) {
+DEVICE real interpolate2D(real x1, real x2, real y1, real y2, real x, real y,
+                            real fx1y1, real fx2y1, real fx1y2, real fx2y2) {
+    real fxy1 = (x2 - x) / (x2 - x1) * fx1y1 + (x - x1) / (x2 - x1) * fx2y1;
+    real fxy2 = (x2 - x) / (x2 - x1) * fx1y2 + (x - x1) / (x2 - x1) * fx2y2;
+
+    return (y2 - y) / (y2 - y1) * fxy1 + (y - y1) / (y2 - y1) * fxy2;
+}
+
+DEVICE void getBounds(real rho, int bounds[]) {
```

Comment:
Same comments here about using an array to return values.  It's more confusing, leads to more complicated code, and will be less efficient if the function doesn't get inlined.